pybuildkite/client.py

```diff
@@ -72,7 +74,10 @@ def request(
             return response
         if method == "DELETE":
             return response.ok
-        if headers == None or headers.get("Accept") == "application/json":
+        if (
+            number_of_passed_in_headers == 0
+            or headers.get("Accept") == "application/json"
+        ):
```

Comment:
I _think_ this is right? Let me know if not.

Reply:
This would work for sure. But now I'm trying to think of an example where we need headers to be `None` and I'm not sure if we do. Maybe instead of allowing headers to be None and working around that, we can just do a simple check at the beginning of the method to see if `headers == None` and throw some `HeadersCannotBeNone` exception? Then all we'd need to check here is the `accept` header like we are